src/public/js/tradeshift-api.js

```diff
@@ -1,9 +1,29 @@
+/*global Q*/
 (function() {
 	var resultArea = document.getElementById('js-api-results');
 
+	function _request(method, path, body) {
+		var host = 'http://localhost:1337/';
+		return Q.Promise(function(resolve, reject) {
+			var xhr = new XMLHttpRequest();
+			xhr.open(method, encodeURI([host, path].join('')), true);
+			xhr.setRequestHeader('Content-Type', 'application/json');
+			xhr.setRequestHeader('Accept', 'application/json');
+			xhr.onload = function() {
+				if (xhr.status === 200) {
+					resolve(xhr.responseText ? JSON.parse(xhr.responseText) : null);
+				} else {
+					reject(xhr.status);
+				}
+			};
+			xhr.send(JSON.stringify(body));
```

Comment:
I can recommend [superagent](https://github.com/visionmedia/superagent) for http request. Works on server + client.
(just a heads up - no code change required)